curriculum/challenges/english/blocks/lecture-working-with-nodejs-and-event-driven-architecture/68ff6d39d7ed9ef36ff281f3.md

```diff
@@ -0,0 +1,433 @@
+---
+id: 68ff6d39d7ed9ef36ff281f3
+title: How can you Install Node on your Computer?
+challengeType: 19
+dashedName: how-can-you-install-node-on-your-computer
+---
+
+# --description--
+
+Now you’re ready to take your first steps into the world of Node.js by installing it on your computer. But first, before you install Node.js, we recommend installing a tool called NVM (Node Version Manager). As its name implies, NVM is a tool for managing multiple versions of Node.js on the same computer.
+
+Setting up this tool from the start will help you in the future because you might need to switch between different versions of Node.js for different projects. With NVM, you can do this very easily.
+
+Let’s see how you can install NVM on macOS/Linux and Windows, and how you can use it to install Node.js.
+
+## How to Install NVM on macOS/Linux
+
+On macOS and Linux, you can install NVM directly through a shell script.
+
+But for macOS, note that the official documentation for NVM mentions that “you need to manually install the Xcode command line tools before running the install script, otherwise, it'll fail.”
+
+On macOS, you can install Xcode command line tools by running this command in your terminal:
+
+```bash
+xcode-select --install
+```
+
+Once the installation is finished, you can check the version number and verify the installation with this command:
+
+```bash
+xcode-select --version
+```
+
+Then, it’s time to install NVM. You can download and execute the NVM installation script by running one of these commands:
+
+```bash
+curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.3/install.sh | bash
+```
+
+```bash
+wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.3/install.sh | bash
+```
+
+To get the latest version of NVM at the time when you are reading this article, please refer to the official NVM documentation: `https://github.com/nvm-sh/nvm?tab=readme-ov-file#install--update-script`. There, you will find these commands with the latest version number.
+
+You might need to restart your terminal when the process is completed to load the new command and the new configuration.
+
+Then, run this command to confirm that NVM was installed correctly and to check your current version:
+
+```bash
+nvm -v
+```
+
+You should see your version of NVM in the output.
+
+## How to install Node.js with NVM on macOS/Linux
+
+Now that you have NVM installed on macOS/Linux, you can install Node.js.
+
+To install the latest Long-Term Support (LTS) version of Node.js, run this command:
+
+```bash
+nvm install --lts
+```
+
+Notice the `--lts` flag. This flag indicates that you want to install the current Long-Term Support (LTS) version of Node.js. This version is usually recommended because it prioritizes stability, reliability, and security over new or experimental features. It is thoroughly tested and ready for production. It’s also guaranteed to have longer support periods with a focus on bug fixes and security patches.
```

Comment:
```suggestion
Notice the `--lts` flag. This flag indicates that you want to install the current Long-Term Support (LTS) version of Node.js. This version is usually recommended because it prioritizes stability, reliability, and security over new or experimental features. It is thoroughly tested and ready for production. It's also guaranteed to have longer support periods with a focus on bug fixes and security patches.
```